contracts/crosschain/Crosschain.sol

```diff
@@ -0,0 +1,147 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.7.1;
+pragma experimental ABIEncoderV2;
+
+import "./TransferDefinition.sol";
+import "../lib/LibChannelCrypto.sol";
+
+/// @title CrosschainTransfer
+/// @author Connext <support@connext.network>
+/// @notice This contract burns the initiator's funds if a mutually signed
+///         transfer can be generated
+
+contract CrosschainTransfer is TransferDefinition {
+    using LibChannelCrypto for bytes32;
+
+    struct TransferState {
+        address initiator;
+        address responder;
+        bytes32 data;
+        uint256 nonce; // Included so that each transfer commitment has a unique hash.
+        uint256 fee;
+        address callTo;
+        bytes callData;
+        bytes32 lockHash;
+    }
+
+    struct TransferResolver {
+        bytes initiatorSignature;
+        bytes responderSignature;
+        bytes32 preImage;
+    }
+
+    // Provide registry information.
+    string public constant override Name = "CrosschainTransfer";
+    string public constant override StateEncoding =
+        "tuple(bytes initiatorSignature, address initiator, address responder, bytes32 data, uint256 nonce, uint256 fee, address callTo, bytes callData, bytes32 lockHash)";
+    string public constant override ResolverEncoding =
+        "tuple(bytes responderSignature, bytes32 preImage)";
+
+    function EncodedCancel() external pure override returns (bytes memory) {
+        TransferResolver memory resolver;
+        resolver.responderSignature = new bytes(65);
+        return abi.encode(resolver);
+    }
+
+    function create(bytes calldata encodedBalance, bytes calldata encodedState)
+        external
+        pure
+        override
+        returns (bool)
+    {
+        // Get unencoded information.
+        TransferState memory state = abi.decode(encodedState, (TransferState));
+        Balance memory balance = abi.decode(encodedBalance, (Balance));
+
+        // Ensure data and nonce provided.
+        require(state.data != bytes32(0), "CrosschainTransfer: EMPTY_DATA");
+        require(state.nonce != uint256(0), "CrosschainTransfer: EMPTY_NONCE");
+
+        // Initiator balance must be greater than 0 and include amount for fee.
+        require(
+            balance.amount[0] > 0,
+            "CrosschainTransfer: ZER0_SENDER_BALANCE"
+        );
+        require(
+            state.fee <= balance.amount[0],
+            "CrosschainTransfer: INSUFFICIENT_BALANCE"
+        );
+
+        // Recipient balance must be 0.
+        require(
+            balance.amount[1] == 0,
+            "CrosschainTransfer: NONZERO_RECIPIENT_BALANCE"
+        );
+
+        // Valid lockHash to secure funds must be provided.
+        require(
+            state.lockHash != bytes32(0),
+            "CrosschainTransfer: EMPTY_LOCKHASH"
+        );
+
+        // Update state.
+        balance.amount[1] = balance.amount[0];
```

Comment:
Create should not be changing anything in the state. This is a pure function that just returns a boolean, so any state changes aren't persisted anywhere.